rapid7_insight_agent/icon_rapid7_insight_agent/util/graphql_api/api_connection.py

```diff
@@ -0,0 +1,345 @@
+
+    def get_agent(self,
+                  agent_input: str
+                  ) -> dict:
+        """
+        Find an agent based on a Mac Address, IP, or Hostname
+
+        :param agent_input: string
+        :return:
+        """
+
+        agent_type = agent_typer.get_agent_type(agent_input)
+        agent = self._get_agent(agent_input, agent_type)
+        return agent
+
+    def quarantine(self,
+                   advertisement_period: int,
+                   agent_id: str
+                   ) -> bool:
+        """
+        quarantine action on a given agent ID
+
+        :param advertisement_period: int (Amount of time in seconds to try to take the quarantine action)
+        :param agent_id: string
+
+        :return: boolean
+        """
+        quarantine_payload = {
+            "query": "mutation( $orgID:String! $agentID:String! $advPeriod:Long! ) { quarantineAssets( orgId:$orgID assetIds: [$agentID] input: {advertisementPeriod: $advPeriod} ) { results { assetId failed } } }",
+            "variables": {
+                "orgID": self.org_key,
+                "agentID": agent_id,
+                "advPeriod": advertisement_period
+            }
+        }
+
+        results_object = self._post_payload(quarantine_payload)
+        failed = results_object.get("data").get("quarantineAssets").get("results")[0].get("failed")
+        return not failed
+
+    def unquarantine(self,
+                     agent_id: str
+                     ) -> bool:
+        """
+        unquarantine action on a given agent ID
+        :param agent_id: string
+
+        :return: boolean
+        """
+        unquarantine_payload = {
+            "query": "mutation( $orgID:String! $agentID:String!) { unquarantineAssets( orgId:$orgID assetIds: [$agentID] ) { results { assetId failed } } }",
+            "variables": {
+                "orgID": self.org_key,
+                "agentID": agent_id
+            }
+        }
+
+        results_object = self._post_payload(unquarantine_payload)
+        failed = results_object.get("data").get("unquarantineAssets").get("results")[0].get("failed")
+        return not failed
+
+    def get_agent_status(self,
+                         agent_id: str
+                         ) -> dict:
+        """
+        This will get agent status information from a specified agent
+
+        :param agent_id: string
+        :return: dict
+        """
+        payload = {
+            "query": "query( $orgID: String! $agentID: String! ) { assets( orgId: $orgID ids: [$agentID] ){ agent { id quarantineState{ currentState } agentStatus } } }",
+            "variables": {
+              "orgID": self.org_key,
+              "agentID": agent_id
+            }
+        }
+
+        results_object = self._post_payload(payload)
+
+        agent = results_object.get("data").get("assets")[0].get("agent")
+        quarantine_state = agent.get("quarantineState").get("currentState")
+        agent_status = agent.get("agentStatus")
```

Comment:
What happens if there are no agents? Will the `[0]` index lookup fail with an IndexError?

Reply:
Add try / except key error.